runtime/etc/Makefile.comm-gasnet

```diff
@@ -58,7 +58,10 @@ LD = $(GASNET_LD)   # GASNet chose C++ linker so stick with it
 else ifeq ($(GASNET_LD),$(GASNET_CC))
 LD = $(GASNET_CXX)  # GASNet chose C, so switch to C++
 else ifeq ($(GASNET_LD_REQUIRES_MPI),1)
-LD = mpicxx         # GASNet chose mpicc, so we'll choose mpicxx
+ifeq (, $(MPI_CXX))
+MPI_CXX = mpicxx    # GASNet defaults to mpicc, so we'll default to mpicxx
```

Comment:
My intuition would be to replicate similar logic in Makefile.gnu and Makefile.intel (setting the variables to the Intel equivalents) such that the compiler Makefiles are establishing what they expect MPI_CC and MPI_CXX to be, and then this can serve as a fallback for any compiler that doesn't set them.  In saying that, I'm assuming that Makefile.gnu and Makefile.intel have long-since been included before this logic such that that would work.

On second thought, though, I'd probably only add it to Makefile.intel since there's nothing in particular about gnu that sets MPI_CXX to be mpicxx (I think it's more that it's using the default name as well?).  So maybe the logic I'd really expect to see in Makefile.intel is:

```
ifeq (, $(MPI_CC))
MPI_CC=mpiicc
endif
 ifeq (, $(MPI_CXX))
MPI_CXX=mpiicpc
endif
```

which would prevent the default setting of MPI_CXX that you've put here from firing and not require a user override?

Reply:
Yeah, good idea. Pushed a commit with that change. Tested that it works on they cray-cs I'm using to repro this, and verified that this does not cause any issues with PrgEnv-intel on a cray-xc.